cpp/src/arrow/engine/substrait/serde.cc

```diff
@@ -56,7 +56,7 @@ Status ParseFromBufferImpl(const Buffer& buf, const std::string& full_name,
   if (message->ParseFromZeroCopyStream(&buf_stream)) {
     return Status::OK();
   }
-  return Status::IOError("ParseFromZeroCopyStream failed for ", full_name);
+  return Status::Invalid("ParseFromZeroCopyStream failed for ", full_name);
```

Comment:
It seemed odd that it failed with an `IOError` given that the `ArrayInputStream` was built from already in-memory Buffer, so no `IO` was involved.